pkg/xds/generator/dns_generator_test.go

```diff
@@ -9,66 +9,83 @@ import (
 	. "github.com/onsi/gomega"
 
 	mesh_proto "github.com/kumahq/kuma/api/mesh/v1alpha1"
+	core_plugins "github.com/kumahq/kuma/pkg/core/plugins"
 	core_mesh "github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
 	model "github.com/kumahq/kuma/pkg/core/xds"
 	xds_types "github.com/kumahq/kuma/pkg/core/xds/types"
+	core_rules "github.com/kumahq/kuma/pkg/plugins/policies/core/rules"
```

Comment:
NO let's not do these crazy imports. You are importing something very specific in a very generic package. This is why our linters and linkers take forever. 

You should be able to just unit test the generic behaviour fo dynconf here. This will create a nice small test that doesn't need to evolve when implementations change. That's "write once, works forever" which is the core of engineering.